### PR TITLE
core-sdk Code refactoring

### PR DESCRIPTION
**Pull request by Nicke-lucky**
Refer irisnet/irishub#2626

Split core-sdk from irishub-sdk-go

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,12 @@ vendor
...
 .idea
+
+# Test binary, built with `go test -c`
+#*.test
+
+# Dependency directories (remove the comment below to include it)
+# vendor/
+*.DS_Store
+*.vscode
+*.idea
```
Comment (chengwenxi) [inline .gitignore:13]: These already exist.

File: core-sdk/IRISHUB-SDK-Go重构设计方案.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+
```
Comment (chengwenxi) [inline core-sdk/IRISHUB-SDK-Go重构设计方案.md:1]: remove this spec

**Comment by Nicke-lucky** [comment]
irishub-sdk-go.  refactor

**Comment by Nicke-lucky** [comment]
Related issues     https://github.com/irisnet/irishub/issues/2626

**Pull request closed without merging**